quil-rs/src/program/analysis/stats/mod.rs

```diff
@@ -66,15 +68,41 @@ impl InstructionsSource for Program {
     }
 }
 
+impl<'p> InstructionsSource for BasicBlock<'p> {
+    type QubitSet = HashSet<&'p Qubit>;
+
+    fn is_empty(&self) -> bool {
+        self.instructions().is_empty()
+    }
+
+    fn len(&self) -> usize {
+        self.instructions().len()
+    }
+
+    fn body_instructions(&self) -> impl Iterator<Item = &Instruction> + '_ {
+        self.instructions().iter().copied()
+    }
+
+    fn get_used_qubits(&self) -> &Self::QubitSet {
+        todo!()
+    }
+}
+
 struct InstructionBlockWithQubitSet<'a, 'b> {
-    block: &'a InstructionBlock<'b>,
+    block: &'a BasicBlock<'b>,
     qubits: HashSet<&'a Qubit>,
 }
 
-impl<'a, 'b> From<&'a InstructionBlock<'b>> for InstructionBlockWithQubitSet<'a, 'b> {
-    fn from(block: &'a InstructionBlock<'b>) -> Self {
+impl<'a, 'b> From<&'a ScheduledBasicBlock<'b>> for InstructionBlockWithQubitSet<'a, 'b> {
```

Comment:
Oh, `InstructionBlockWithQubitSet` was just a hack to let `InstructionBlock` work as an instruction-source.